### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by dennisklein**
* Add verify output in section `Compiler/Install`
* Add section `Structure of this repo`
* Add troubleshooting entry about DDS + external non-system compiler

---

Note: I deleted the `-b <release>` argument from the first section on `git clone ...`. If you intend to add tags/branches for the various releases and intend the user to check them out in their git working tree, then you might want to re-add it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,13 +147,71 @@ Verify the installation
...
 
+## [Troubleshooting](docs/troubleshooting.md)
+
+## Structure of this repo
+
+Releases are defined as a Spack environment in the directory [`env/<release>/<variant>.yaml`](env/).
+
+`repo.yaml` + `packages/` constitute the FairSoft Spack package repository.
+
+`config/` contains some general configuration changes needed on some systems, please load it with `spack -C ./config install`.
```
Comment (ChristianTackeGSI) [inline README.md]: Maybe:

```suggestion

`config/` contains some general configuration changes needed on some systems, please use it with `spack -C ./config …`.
```

File: README.md [hunk 2]
```diff
@@ -146,13 +147,71 @@ Verify the installation
...
+
+`spack/` is a git submodule which references the upstream spack git repo which currently contains both the Spack software and the Spack builtin package repository. Because some of the packages a FairSoft release pins down come from the Spack builtin repository, the idea is to pin it, too, to provide a reproducible build experience which does not depend on the various possible combination of Spack and FairSoft.
```
Comment (ChristianTackeGSI) [inline README.md]: Maybe add this?
> The referenced submodule might also contain some extra patches, that we were not yet able to integrate into spack itself.

File: README.md (hunk 1 above)
Comment (dennisklein) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (dennisklein) [reply README.md]: done

**Pull request merged**